src/System.Text.Json/tests/JsonNumberTestData.cs

```diff
@@ -0,0 +1,214 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Buffers;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using Newtonsoft.Json;
+
+namespace System.Text.Json.Tests
+{
+    internal class JsonNumberTestData
+    {
+        public static List<int> Ints { get; set; }
+        public static List<long> Longs { get; set; }
+        public static List<uint> UInts { get; set; }
+        public static List<ulong> ULongs { get; set; }
+        public static List<float> Floats { get; set; }
+        public static List<double> Doubles { get; set; }
+        public static List<decimal> Decimals { get; set; }
+        public static byte[] JsonData { get; set; }
+
+        static JsonNumberTestData()
+        {
+            var random = new Random(42);
+
+            const int numberOfItems = 1_000;
+
+            // Make sure we have 1_005 values in each numeric list.
+
+            #region generate ints
+            Ints = new List<int>
+            {
+                0,
+                12345,
+                -12345,
+                int.MaxValue,
+                int.MinValue
+            };
+            for (int i = 0; i < numberOfItems; i++)
+            {
+                int value = random.Next(int.MinValue, int.MaxValue);
+                Ints.Add(value);
+            }
+            #endregion
+
+            #region generate longs
+            Longs = new List<long>
+            {
+                0,
+                12345678901,
+                -12345678901,
+                long.MaxValue,
+                long.MinValue
+            };
+            for (int i = 0; i < numberOfItems; i++)
+            {
+                long value = random.Next(int.MinValue, int.MaxValue);
+                if (value < 0)
+                    value += int.MinValue;
+                else
+                    value += int.MaxValue;
+                Longs.Add(value);
+            }
+            #endregion
+
+            #region generate uints
+            UInts = new List<uint>
+            {
+                uint.MinValue,
+                uint.MaxValue,
+                12345,
+                // next two values are just to satisfy requirement of having 5 values in the list
+                67890,
```

Comment:
Right since we can't have negative values, and `uint.MinValue == 0`. Good comment :)